overlord/snapstate/snapstate_test.go

```diff
@@ -9046,3 +9046,167 @@ func (s *snapmgrTestSuite) TestRefreshCandidatesMergeFlags(c *C) {
 		},
 	})
 }
+
+func (s *snapmgrTestSuite) TestCleanSnapDownloadsSnapNotExist(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// non-existing snaps should be ignored
+	err := snapstate.CleanSnapDownloads(s.state, "some-snap")
+	c.Check(err, IsNil)
+}
+
+func (s *snapmgrTestSuite) TestCleanSnapDownloads(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	initSnapDownloads := func(revisions []snap.Revision) func() {
+		c.Assert(os.MkdirAll(dirs.SnapBlobDir, 0755), IsNil)
+		for _, rev := range revisions {
+			fileName := fmt.Sprintf("some-snap_%s.snap", rev)
+			c.Assert(ioutil.WriteFile(filepath.Join(dirs.SnapBlobDir, fileName), nil, 0644), IsNil)
+		}
+		return func() {
+			c.Assert(os.RemoveAll(dirs.SnapBlobDir), IsNil)
+		}
+	}
+
+	// 1. revisions in snap's sequence are kept
+	restore := initSnapDownloads([]snap.Revision{snap.R(1), snap.R(2), snap.R(3)})
```

Comment:
This test should use revisions with more than 1 digit. Changing 1 to 11 here and in line 9090, triggers the bug I mentioned in the previous comment